python-package/setup.py

```diff
@@ -40,20 +42,31 @@ def copy_files_helper(folder_name):
     if not os.path.isfile('./_IS_SOURCE_PACKAGE.txt'):
         copy_files_helper('include')
         copy_files_helper('src')
+        copy_files_helper('windows')
         if use_gpu:
             copy_files_helper('compute')
         distutils.file_util.copy_file("../CMakeLists.txt", "./lightgbm/")
         distutils.file_util.copy_file("../LICENSE", "./")
 
 
 def clear_path(path):
-    contents = os.listdir(path)
-    for file in contents:
-        file_path = os.path.join(path, file)
-        if os.path.isfile(file_path):
-            os.remove(file_path)
-        else:
-            shutil.rmtree(file_path)
+    if os.path.isdir(path):
+        contents = os.listdir(path)
+        for file in contents:
+            file_path = os.path.join(path, file)
+            if os.path.isfile(file_path):
+                os.remove(file_path)
+            else:
+                shutil.rmtree(file_path)
+
+
+def silent_call(cmd):
```

Comment:
I am not so sure about what's this function for, why makes it `silent`?

Reply:
@wxchan This PR was conceived as fix to #874

Reply:
@StrikerRUS ok, got it.

Reply:
Can this function be designed like `def silent_call(cmd, raise_error=True, error_msg='')`? In this case, if cmake_cmd fails, you don't need to call make_cmd then.

Reply:
@wxchan 
It already has described behavior. In case of any error it returns `1` and you should just check returned value.

Do you mean that I should change this code
```
    status = silent_call(cmake_cmd + ["-G", "MinGW Makefiles"])
    status += silent_call(["mingw32-make.exe", "_lightgbm"])
        if status != 0:
            raise Exception('Please install CMake and MinGW first')
```
to
```
status = silent_call(cmake_cmd + ["-G", "MinGW Makefiles"])
if status != 0:
    raise Exception('Please install CMake first')
status = silent_call(["mingw32-make.exe", "_lightgbm"])
if status != 0:
    raise Exception('Please install MinGW first')
```

Reply:
yes, I think it's better, and you can move raise Exception into function to avoid duplicates, but it's trivial, up to you.

Reply:
@wxchan Moving raising Exception into the function will cause dozens of `try/except` into the main code and will make code messy.

Reply:
doesn't sth like below works? why need try/except in main code? not sure, LGTM now, it's up to you.
```
def silent_call(cmd, raise_error=True, error_msg='')
	try:
		// do sth
		return 0
	except:
		if raise_error:
			raise Exception(error_msg)
		return 1

silent_call(cmake_cmd, error_msg='Please install CMake first')
```

Reply:
@wxchan It seems you are right! `raise_error` will help to deal with such situation
`if status == 0 and os.path.exists(lib_path):`